pkg/generate.go

```diff
@@ -182,20 +183,50 @@ func GenerateOverlays(fs afero.Afero, gitOpsFolder string, outputFolder string,
 
 	deploymentPatch := generateDeploymentPatch(options, imageName, containerName, namespace)
 
+	resources := map[string]interface{}{
+		deploymentPatchFileName: deploymentPatch,
+	}
+
 	k.AddResources("../../base")
 	k.AddPatches(deploymentPatchFileName)
-	if componentGeneratedResources == nil {
-		componentGeneratedResources = make(map[string][]string)
-	}
 	componentGeneratedResources[options.Name] = append(componentGeneratedResources[options.Name], deploymentPatchFileName)
 
+	var route *routev1.Route
+	var ingress *networkingv1.Ingress
+
+	// Create an ingress if its a Kubernetes cluster, route if its an OpenShift cluster
+	if clusterInfo.IsKubernetesCluster {
+		if len(options.KubernetesResources.Ingresses) == 0 && options.TargetPort != 0 {
+			// If no Ingresses were provided and TargetPort is not 0, generate the Ingress
```

Comment:
I wonder if we could simplify things here and leave the route/ingress generation just to the gitops-generator? Rather than passing in our own ingress/route via the `KubernetesResources` option.

The information that would need to be passed is really only the following:
- Target port (`spec.targetPort` vs. endpoint port)
- Ingress domain
- Component's service name

Reply:
The thing is, this was an option, but what happens when someone doesn't provide a devfile.kubernetes.endpoint? And if a devfile endpoint is not provided and someone provides route/ingress in their yaml in devfile?

So i guess this approach gitops gen library can either generate one route/ingress if finds no information passed in or just use the one provided 🤔 and it HAS controller responsiblity to parse the devfile and figure out what ingress/route is present in it and pass in the whole resource.

Reply:
I know its not really ideal but I would definitely like to rework some of these logic but maybe after service preview.

Reply:
We discussed this on Friday in a F2F meeting, any improvements/updates can be made after Service Preview. I will open an issue and assign myself for making these iterative improvements to Gitops generation